### DISCUSSION TITLE
昨天遇到一个路由优先级配置问题导致的事故，分享下，看下有没有什么好的方案

### DISCUSSION THREAD

**Discussion by Lewisyixin** [apache/apisix#7612] [Q&A]
简化来说我们有两条路由规则，两个域名的根路径转发到了各自的业务，并且各自加了域名匹配
1. /*     Host: www.test1.com
2. /*     Host: www.test2.com

然后修改第一条路由的时候，不小心把Host规则删掉了，变成了
1. /*    
2. /*    Host: www.test2.com

这个时候路由1的优先级比路由2高，导致了原本想走 www.test2.com的路由全都转到了 路由1。

总结下，同一个apisix肯定会维护多个域名转发，每个域名大概率也会有一个/路由转给静态资源，不管怎么配置，一旦把优先级高的跟路由的域名规则不小心删除，就会覆盖所有的其他跟路由，感觉这个有一定的风险，不知道大家有没有遇到类似的问题，有没有什么好的规避方案？

**Comment by Lewisyixin** [comment]
如果host为空的时候，改为匹配不到任何，而不是能匹配任何  可行吗？

**Reply by Lewisyixin** [reply]
找到方案了 可以吧host+uri作为索引条件
https://apisix.apache.org/zh/docs/apisix/terminology/router/

**Reply by tokers** [reply]
I think another way is giving users a hint, for instance, when we execute on APISIX dashboard, we can give users an alert: "Your route doesn't contain a Host, are you sure you want to apply the changes?".

**Reply by Lewisyixin** [reply]
是的，dashboard上增加提示确实是可以减少此类事故发生，但是归根结底避免的方式还是得调整这个路由匹配方式。 如果将当前优先级最高的路由的host条件去掉将会导致所有同路由的访问，影响的是所有站点，这个风险确实太高了，生产环境需要一个100%避免这种重大事故的方式。目前看host+uri匹配方式可以满足这种需求

**Comment by crystalww** [comment]
同样遇到这个问题，有两条路由：
1. /*                                        Host:  xxxx
2. /notebook/jupyter/1/lab    不设置Host

结果想访问1的path都请求到了路由2上。   文档中也没找到关于设置host与不设置host的匹配优先级说明啊

**Reply by tokers** [reply]
Try to read https://apisix.apache.org/docs/apisix/router-radixtree/. You need to use another router `radixtree_host_uri`, instead of the default `radixtree_uri`.
